src/reporter.ts

```diff
@@ -153,32 +176,35 @@ export function getPrioritizedSpans(spans: any[], maxSendBytes: number): any[] {
     const spanSize = getJSONBase64Size(spans[index]);
     const spanMetadataSize = spansToSendSizes[index];
 
-    if (currentSize + spanSize - spanMetadataSize < maxSendBytes) {
+    if (currentSize + spanSize - spanMetadataSize < maxSendBytes - sendEnrichmentSpanSizeBuffer) {
       spansToSend[index] = spans[index];
       currentSize += spanSize - spanMetadataSize;
     }
   }
 
-  const spansDropped = spans.length - Object.keys(spansToSend).length;
+  let finalSpansToSend = Object.values(spansToSend);
+  const spansDropped = spans.length - finalSpansToSend.length;
   if (spansDropped > 0) {
-    SpansContainer.recordDroppedSpan(DroppedSpanReasons.SPANS_SENT_SIZE_LIMIT, false, spansDropped);
     logger.info(`Dropped ${spansDropped} spans due to size limit of total spans sent to lumigo`);
 
-    // Update the end span with the new recorded drops
-    let endSpanFound = false;
-    for (let index = 0; index < Object.values(spansToSend).length; index++) {
-      if (spansToSend[index].type === FUNCTION_SPAN) {
-        spansToSend[index].droppedSpansReasons = SpansContainer.getDroppedSpansReasons();
-        endSpanFound = true;
-        break;
+    if (addEnrichmentSpan) {
+      // If we are not adding an enrichment span then no need to record these dropped spans, they might be re-sent in a later stage in the invocation
+      SpansContainer.recordDroppedSpan(
+        DroppedSpanReasons.SPANS_SENT_SIZE_LIMIT,
+        false,
+        spansDropped
+      );
+      if (!enrichmentSpansDropped) {
+        finalSpansToSend = addOrUpdateEnrichmentSpan(finalSpansToSend);
+      } else {
+        logger.warn(
+          'Enrichment span was dropped due to size limit of total spans sent to lumigo, so dropped spans counts might be missing'
```

Comment:
don't we want to give the enrichment span priority?

Reply:
We do, but the function span has a higher priority.
We shouldn't reach this point, the only scenario when we will reach this is if the metadata of the function start + end spans + the metadata of the enrichment span is over the max send limit.

In any case I rather get the lambda invocation fact with no enrichment data then only enrichment data with no lambda invocation fact